File: indexer/models/email.go
package models

import "time"

// Email represents an email structure to be indexed
type Email struct {
	Body                    string    `json:"body"`
	CFolder                 string    `json:"c_folder"`
	CC                      string    `json:"cc"`
	ContentTransferEncoding string    `json:"content_transfer_encoding"`
	ContentType             string    `json:"content_type"`
	Date                    time.Time `json:"date"`
	From                    string    `json:"from"`
	MessageID               string    `json:"message_id"`
	MimeVersion             string    `json:"mime_version"`
	Replies                 []string  `json:"replies"`
	Sent                    time.Time `json:"sent"`
	Subject                 string    `json:"subject"`
	To                      string    `json:"to"`
	XBcc                    string    `json:"x_bcc"`
	XCc                     string    `json:"x_cc"`
	XFileName               string    `json:"x_file_name"`
	XFrom                   string    `json:"x_from"`
	XOrigin                 string    `json:"x_origin"`
	XTo                     string    `json:"x_to"`
}

type Replies struct {
	From    string `json:"from"`
	To      string `json:"to"`
	Send    string `json:"send"`
	Subject string `json:"subject"`
	Cc      string `json:"cc"`
	Content string `json:"content"`
}

File: email-viewer/src/types/email.ts

export interface Response {
took: number;
timed_out: boolean;
_shards: Shards;
hits: Hits;
}

export interface Shards {
total: number;
successful: number;
skipped: number;
failed: number;
}

export interface Hit {
_index: string;
_type: string;
_id: string;
_score: number;
"@timestamp": string;
_source: Email;
}

export interface Hits {
total: Total;
max_score: number;
hits: Hit[];
}

export interface Email {
"@timestamp": string;
body: string;
c_folder: string;
cc: string;
content_transfer_encoding: string;
content_type: string;
date: string;
date_subemail: string;
from: string;
message_id: string;
mime_version: string;
sent: string;
subject: string;
to: string;
x_bcc: string;
x_cc: string;
x_file_name: string;
x_from: string;
x_origin: string;
x_to: string;
}

export interface Total {
    value: number;

} 